browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -183,6 +142,83 @@ class AdblockCnameResolveHostClient : public network::mojom::ResolveHostClient {
   }
 };
 
+// If `canonical_url` is specified, this will only check if the CNAME-uncloaked
+// response should be blocked. Otherwise, it will run the check for the
+// original request URL.
+EngineFlags ShouldBlockRequestOnTaskRunner(
+    std::shared_ptr<BraveRequestInfo> ctx,
+    EngineFlags previous_result,
+    base::Optional<GURL> canonical_url) {
+  if (!ctx->initiator_url.is_valid()) {
+    return previous_result;
+  }
+  const std::string source_host = ctx->initiator_url.host();
+
+  GURL url_to_check;
+  if (canonical_url.has_value()) {
+    url_to_check = *canonical_url;
+  } else {
+    url_to_check = ctx->request_url;
+  }
+
+  g_brave_browser_process->ad_block_service()->ShouldStartRequest(
+      url_to_check, ctx->resource_type, source_host,
+      &previous_result.did_match_rule, &previous_result.did_match_exception,
+      &previous_result.did_match_important, &ctx->mock_data_url);
+
+  if (previous_result.did_match_important ||
+      (previous_result.did_match_rule &&
+       !previous_result.did_match_exception)) {
+    ctx->blocked_by = kAdBlocked;
+  }
+
+  return previous_result;
+}
+
+void OnShouldBlockRequestResult(
+    bool then_check_uncloaked,
+    scoped_refptr<base::SequencedTaskRunner> task_runner,
+    const ResponseCallback& next_callback,
+    std::shared_ptr<BraveRequestInfo> ctx,
+    EngineFlags result) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  if (ctx->blocked_by == kAdBlocked) {
+    brave_shields::BraveShieldsWebContentsObserver::DispatchBlockedEvent(
+        ctx->request_url, ctx->frame_tree_node_id, brave_shields::kAds);
+  } else if (then_check_uncloaked) {
+    // This will be deleted by `AdblockCnameResolveHostClient::OnComplete`.
+    new AdblockCnameResolveHostClient(std::move(next_callback), task_runner,
+                                      ctx, result);
+    return;
+  }
+  next_callback.Run();
+}
+
+void UseCnameResult(scoped_refptr<base::SequencedTaskRunner> task_runner,
+                    const ResponseCallback& next_callback,
+                    std::shared_ptr<BraveRequestInfo> ctx,
+                    const EngineFlags previous_result,
```

Comment:
arguments passed by value don't need const, just `EngineFlags previous_result` and `base::Optional<std::string> cname`

Reply:
Doesn't it just have the same semantics as you mentioned in https://github.com/brave/brave-core/pull/8704#discussion_r630397773?

At any rate, removed.